EXPERIMENTAL - Factor out PDF module string constants

carlwilson opened the pull request:
- created class for PDF MessageConstants; and
  - factoring out all String constants.

carlwilson committed:
EXPERIMENTAL - Factor out string constants
 - created class for PDF MessageConstants;
 - factoring out all String message constants to MessageConstants; and
 - started, but not finished, factoring out dict keys and property names.

sqash

david-russo commented:
I like how _out_ you've factored them, but at the same time I feel they could be even _further_ out. Like resource-file out. If we had them in such a file, it might also be useful as a source for generating the online documentation... and other language translations... the methods for working with which could all be generalised under jhove-core for the benefit of all modules... but I'm getting _waaay_ ahead of this pull request...

Regarding which, we might want to make a note to change the message constant names for their equivalent IDs once they've been created, to avoid them possibly falling out of sync with their underlying messages.

Anyway, nice work, certainly an improvement :) I see no immediate problems with this, besides having to update the line numbers in the wiki again... Maybe leave this until after the hack day?

jayGattusoNLNZ committed:
Create MessageConstants.java

template based on pdf example

jayGattusoNLNZ committed:
Update MessageConstants.java

jayGattusoNLNZ committed:
Update MessageConstants.java

jayGattusoNLNZ committed:
Create MessageConstants.java

Needed to comment out some of the var defs, (L46, 47, 62, 76) due to poor JAVA skillz.

jayGattusoNLNZ committed:
Create MessageConstants.java

This will need some pruning...

pmay commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/MessageConstants.java:
```diff
@@ -0,0 +1,160 @@
+/**
```
Missing "Invalid RoleMap" error from https://github.com/openpreserve/jhove/blob/release-1.14/jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/StructureTree.java#L249

carlwilson replied:
Yvonne caught another, I'm on it ;)

carlwilson committed:
Merge branch 'integration' into exp-message-constants

carlwilson committed:
Merge remote-tracking branch 'upstream/integration' into exp-message-constants

carlwilson committed:
FIX - Compile errors in message constants
- only PDF messages used in code as we have some test files; and
- checked Black Box testing by breaking a message.

carlwilson committed:
Merge branch 'integration' into exp-message-constants

carlwilson committed:
Merge branch 'integration' into exp-message-constants

carlwilson committed:
Merge remote-tracking branch 'origin/integration' into exp-message-constants
- bumped PDF Module version number;
- finally merged old message constants for PDF module; and
- tracked down some missed strings and properties in `PdfModule` class;

carlwilson committed:
FIX - Codacy warnings
- string repetition in `PdfModule`; and
- Eclipse artifacts into .gitignore.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java:
```diff
@@ -109,21 +110,274 @@
+
+	private static final String DICT_KEY_FONT_DESCRIPTOR = "FontDescriptor";
+	private static final String DICT_KEY_STARTXREF = "startxref";
+	private static final String DICT_KEY_BASE_FONT = base + font;
```
This code would be easier to read as a literal "BaseFont", cutting out the time it takes to confirm the contents of the variables, and allowing the key to be searched. They're also variables that are unlikely to need changing in the future, so I see little benefit in having them as variables. I think the same can be said for most of the variables declared on lines 116 to 121, with the exception of `crop_box`.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java:
```diff
@@ -109,21 +110,274 @@
+	public static final String EXT = ".pdf";
+	
+	private static final String font = "Font";
+	private static final String crop_box = "CropBox";
```
Perhaps `PROP_VAL_CROP_BOX` to match the others?

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/PdfModule.java:
```diff
@@ -109,21 +110,274 @@
+	private static final String PDF_SIG_HEADER = "%" + PDF_VER1_HEADER_PREFIX;
+	private static final String POSTSCRIPT_HEADER_PREFIX = "!PS-Adobe-";
+	private static final String ENCODING_PREFIX = "ENC=";
+    private static final String NO_HEADER = "No PDF header";
```
Consistent space indentation to match the rest/majority of the file.

david-russo commented on jhove-modules/src/main/java/edu/harvard/hul/ois/jhove/module/pdf/Tokenizer.java:
```diff
@@ -55,6 +55,9 @@
     };
 
+    private static final String empty = "";
+    private static final String stream = "stream";
```
`EMPTY`, `STREAM`?

carlwilson committed:
REV - Post review from @david-russo
- minor changes and tweaks.
